### PR TITLE
8212961: [TESTBUG] vmTestbase/nsk/stress/jni/ native code cleanup 

### PR DESCRIPTION
**Pull request by dholmes-ora**
Please review what is primarily a code style cleanup with some old native code:

- add spaces around operators and after commas
- fixed indent and line breaks
- added braces to blocks
- replaced raw malloc with c_malloc (check malloc)
- fixed CE macro using and exception processing in general
  - added different CHECK macro for monitor enter/exit
  - removed some unreachable explicit exception checking code

I staged this as multiple commits to try and make the changes more obvious at each step (and if you hide whitespace changes it simplifies quite a bit), but it may be easier to just look at the end results and see if anything jumps out as still needing changing.

Note: there was no attempt to actually change the logic of any test or rewrite them in a better way - that is out of scope for the cleanup. The only logic changes relates to the CE macro and exception/error checking.

Testing:
- local running of the tests themselves
- tier5 testing in our CI system (in progress).

This may be of interest to @iignatev  and @lmesnik .

Thanks,
David

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8212961](https://bugs.openjdk.java.net/browse/JDK-8212961): [TESTBUG] vmTestbase/nsk/stress/jni/ native code cleanup

### Reviewers
 * [Thomas Stuefe](https://openjdk.java.net/census#stuefe) (@tstuefe - **Reviewer**)
 * [Igor Ignatyev](https://openjdk.java.net/census#iignatyev) (@iignatev - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/4882/head:pull/4882` \
`$ git checkout pull/4882`

Update a local copy of the PR: \
`$ git checkout pull/4882` \
`$ git pull https://git.openjdk.java.net/jdk pull/4882/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 4882`

View PR using the GUI difftool: \
`$ git pr show -t 4882`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/4882.diff">https://git.openjdk.java.net/jdk/pull/4882.diff</a>

</details>

### CODE REVIEW

File: test/hotspot/jtreg/vmTestbase/nsk/stress/jni/libjnistress001.cpp [hunk 1]
```diff
@@ -33,214 +33,221 @@ extern "C" {
...
+      env->CallStaticVoidMethod(clazz, methodID, JNI_FALSE); CE
     }
-    env->MonitorExit(jobj); CE
-    return(env->NewString(javachars->str[index-1],elem_len));
+    //methodID = env->GetStaticMethodID(clazz, halt, haltSig); CE
+    //env->CallStaticVoidMethod(clazz, methodID); CE
+    return tmpstr;
+  }
+  CHECK(env->MonitorExit(jobj));
+  return env->NewString(javachars->str[index - 1], elem_len);
```
Comment (tstuefe) [inline test/hotspot/jtreg/vmTestbase/nsk/stress/jni/libjnistress001.cpp:250]: Should you check for exceptions after NewString?

File: test/hotspot/jtreg/vmTestbase/nsk/stress/jni/libjnistress003.cpp [hunk 2]
```diff
@@ -40,225 +40,230 @@ extern "C" {
...
 
-    for (i=0;i<SIZE;i++) {
-    if (i%2 == 0) boolBuf[i]=JNI_TRUE;
-    else boolBuf[i]=JNI_FALSE;
+  for (i = 0; i < SIZE; i++) {
+    if (i % 2 == 0) {
+      boolBuf[i] = JNI_TRUE;
+    } else {
+      boolBuf[i] = JNI_FALSE;
+    }
```
Comment (tstuefe) [inline test/hotspot/jtreg/vmTestbase/nsk/stress/jni/libjnistress003.cpp:89]: Shorten to `boolBuf[i] = (i % 2) ? JNI_TRUE : JNI_FALSE` ?

**Review by tstuefe — COMMENTED** [review]
Looks good, small remarks inside.

Cheers, Thomas

File: test/hotspot/jtreg/vmTestbase/nsk/stress/jni/libjnistress001.cpp (hunk 1 above)
Comment (dholmes-ora) [reply test/hotspot/jtreg/vmTestbase/nsk/stress/jni/libjnistress001.cpp:250]: It isn't essential as any exception will be thrown in the calling Java code. It won't be as fail-fast as using CE but doesn't seem worth the effort to change.

File: test/hotspot/jtreg/vmTestbase/nsk/stress/jni/libjnistress003.cpp (hunk 2 above)
Comment (dholmes-ora) [reply test/hotspot/jtreg/vmTestbase/nsk/stress/jni/libjnistress003.cpp:89]: Sure :)

**Comment by dholmes-ora** [comment]
Thanks for looking at this @tstuefe !

File: test/hotspot/jtreg/vmTestbase/nsk/stress/jni/libjnistress003.cpp (hunk 2 above)
Comment (dholmes-ora) [reply test/hotspot/jtreg/vmTestbase/nsk/stress/jni/libjnistress003.cpp]: but without the implicit boolean :)

**Review by tstuefe — APPROVED** [review]
Looks good!

**Comment by dholmes-ora** [comment]
Looking for a second review please.

**Review by iignatev — APPROVED** [review]
LGTM

**Comment by dholmes-ora** [comment]
Thanks for the review @iignatev !

/integrate

**Pull request closed without merging**